### PR TITLE
Support billboarding via the `KHR_billboard` extension

### PR DESCRIPTION
**Pull request by j9liu**
## Description

Credit to @david-lively for most of this code.

This PR adds support for billboarding meshes with the `KHR_billboard` extension.

Note that this is really simple support, and we don't properly handle `rotationAxis`, `overlay`, etc. properties. The code to make billboarding work with an arbitrary up-axis was non-trivial, so the `rotationAxis` stuff could sadly become a potential nightmare.

If need be, these can be handled in a future PR.

Here's the show of the material function code:

image

Notably this is toggled with an `if`. Typically `if`s aren't ideal, but all the points of the same mesh will go down the same branch, so this shouldn't be too costly.

## Issue number or link

N/A

## Author checklist

- [X] I have submitted a [Contributor License Agreement](https://github.com/CesiumGS/community/tree/main/CLAs) (only needed once).
- [X] I have done a full self-review of my code.
- [x] I have updated `CHANGES.md` with a short summary of my change (for user-facing changes).
- ~[ ] I have added or updated unit tests to ensure consistent code coverage as necessary.~
- ~[ ] I have updated the documentation as necessary.~

## Testing plan

Reach out to me for a sample tileset featuring the billboard data we've used for testing.

I was hoping to use the [BillboardDemo sample](https://github.com/KhronosGroup/glTF-External-Reference/blob/main/demo/Models/BillboardDemo/glTF/BillboardDemo.gltf) to showcase this, but turns out that the model uses the `rotationAxis` property we have intentionally omitted as part of this PR. So the billboards are not rotating the intended way.

| Good | Bad |
| - | - |
| image | image |

## Reviewer checklist

Thank you for taking the time to review this PR. By approving a PR you are taking as much responsibility for these changes as the author.

As you review, please go through the checklist below:

- [ ] Review and run all parts of the test plan on this branch and verify it matches expectations.
    - [ ] If the issue is a bug please make sure you can reproduce the bug in the main branch and then checkout this branch to make sure it actually solved the issue.
- [ ] Review the code and make sure you do not have any remaining questions or concerns. You should understand the code change and the chosen approach. If you are not confident or have doubts about the code, please do not hesitate to ask questions.
    - [ ] Code should follow the [C++ Style Guide](https://cesium.com/learn/cesium-native/ref-doc/style-guide.html)
- [ ] Review the unit tests and make sure there are no missing tests or edge cases.
- [ ] Review documentation changes and updates to `CHANGES.md` to make sure they accurately cover the work in this PR.
- [ ] Verify that the [Contributor License Agreement](https://github.com/CesiumGS/community/tree/main/CLAs) has been submitted, if needed.

### CODE REVIEW

File: Source/CesiumRuntime/Private/CesiumGltfComponent.cpp [hunk 1]
```diff
@@ -3139,6 +3147,19 @@ PRAGMA_ENABLE_DEPRECATION_WARNINGS
...
+    UMaterialInstanceDynamic* pMaterial,
+    const std::optional<CesiumGltf::ExtensionKhrBillboard>& billboard,
+    const EMaterialParameterAssociation association,
+    const int32 index) {
+
+  pMaterial->SetScalarParameterValueByInfo(
+      FMaterialParameterInfo(FName("isBillboard"), association, index),
+      billboard ? 1.0 : 0.0);
+
+  // Future TODO: support other properties in the KHR_billboard extension.
```
Comment (j9liu) [inline Source/CesiumRuntime/Private/CesiumGltfComponent.cpp:3159]: There are additional properties in `KHR_billboard` ([PR for reference](https://github.com/KhronosGroup/glTF/pull/2522/changes)) that are potentially non-trivial to implement given that we have to work with a varying up-axis across the world. @david-lively put a lot of effort into getting this to work for that already. I think deferring this to a later PR if there is community need / interest is the way to go.

**Comment by azrogers** [comment]
That if statement could be turned into a multiplication, but to be honest I don't know enough about GPU scheduling to say whether that would be faster. It definitely seems like performance isn't a problem right now anyways so nothing to worry about. Thanks @j9liu!
